tests/datastore/modifycommontests.nim

```diff
@@ -108,17 +108,10 @@ proc modifyTests*(
     proc returningAux(_: ?seq[byte]): Future[(?seq[byte], seq[byte])] {.async.} =
       return (seq[byte].none, @[byte 123])
 
-    let res = await ds.modifyGet(key, returningAux)
+    let aux = (await ds.modifyGet(key, returningAux)).tryGet()
 
-    if multiAux:
-      check:
-        res.errorOption.map((err) => err.msg) == none(string)
-      for b in res |? @[]:
-        check:
-          b == 123.byte
-    else:
-      check:
-        res == success(@[byte 123])
+    check:
+      aux == (123.byte).repeat(dsCount)
```

Comment:
I don't understand the changes in this file at all. Please explain.

Reply:
So there's a case of `TieredDatastore` that need this dedicated handling here. `TieredDatastore` is defined as a composition of few other stores.
```
TieredDatastore* = ref object of Datastore
    stores: seq[Datastore]
```

When calling methods like `put` or `modify`, all stores gets updated.

API has also another method called `modifyGet` that not only update the value but also return some auxiliary value (could be anything that developer defines like for example previous value before update).

When we have multiple stores, we have multiple auxiliary values and they have to be combined somehow. The arbitrary choice I've made some time ago is to simply concatenate these values.

In test above the single auxiliary value we return is just `@[byte 123]` (a sequence with a single byte). When we concatenate multiple auxiliary values we get `@[byte 123, 123, 123, ...]`, which is the same as `(123.byte).repeat(n)`. `n` is just number of stores we use (which is 2 in this case).

I've decided to make this test consistent with what's in the `typedscommontests.nim` and use `dsCount` instead of `multiAux` parameter, however the previous version would also work.

Reply:
OK, I understand.
I don't really like it, because this behavior isn't obvious or what a reader would expect from the names. But... I haven't come up with anything better or a different solution. If this is just caused by the TieredDatastore, and we are not using the TieredDatastore anywhere (we aren't), maybe it would be cleaner to drop it entirely?

Reply:
I kind of agree that maybe it would be better to drop the `TieredDatastore` entirely since it's not used anywhere. However such change is not part of the scope for this PR.